File: src/dataset/javascript/robot-simulator_exercism-ThomasZumsteg/openAI/ChatGPT4_robot-simulator_v3.js
class Robot {
    constructor() {
        this.bearing = null;
        this.coordinates = [0, 0];
    }

    static directions = ['north', 'east', 'south', 'west'];
    static vectors = { north: [0, 1], east: [1, 0], south: [0, -1], west: [-1, 0] };
    static instructionsMap = { L: 'turnLeft', R: 'turnRight', A: 'advance' };

    orient(direction) {
        if (!Robot.directions.includes(direction)) throw "Invalid Robot Bearing";
        this.bearing = direction;
    }

    turnRight() {
        this.bearing = Robot.directions[(Robot.directions.indexOf(this.bearing) + 1) % 4];
    }

    turnLeft() {
        this.bearing = Robot.directions[(Robot.directions.indexOf(this.bearing) + 3) % 4];
    }

    at(x, y) {
        this.coordinates = [x, y];
    }

    advance() {
        const [dx, dy] = Robot.vectors[this.bearing];
        this.coordinates[0] += dx;
        this.coordinates[1] += dy;
    }

    instructions(instructionList) {
        return [...instructionList].map(instruction => Robot.instructionsMap[instruction]);
    }

    place({ direction, x, y }) {
        this.orient(direction);
        this.at(x, y);
    }

    evaluate(instructionList) {
        this.instructions(instructionList).forEach(action => this[action]());
    }
}

module.exports = Robot;
exports.directions = Robot.directions;

File: src/dataset/python/robot-simulator_exercism-ThomasZumsteg/openAI/ChatGPT4_robot_simulator_v3.py
NORTH, EAST, SOUTH, WEST = 0, 1, 2, 3

class Robot:
    _valid_commands = {
        'R': 'turn_right',
        'L': 'turn_left',
        'A': 'advance',
    }

    _direction_moves = {
        NORTH: (0, 1),
        EAST: (1, 0),
        SOUTH: (0, -1),
        WEST: (-1, 0),
    }

    def __init__(self, bearing=NORTH, x=0, y=0):
        self.bearing = bearing
        self.coordinates = (x, y)

    def turn_right(self):
        self.bearing = (self.bearing + 1) % 4

    def turn_left(self):
        self.bearing = (self.bearing - 1) % 4

    def advance(self):
        dx, dy = self._direction_moves[self.bearing]
        x, y = self.coordinates
        self.coordinates = (x + dx, y + dy)

    def simulate(self, commands):
        for c in commands:
            getattr(self, self._valid_commands[c])()